Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift

```diff
@@ -350,25 +353,28 @@ public final class PackageInfoMapper: PackageInfoMapping {
             }
         )
 
-        let targets = try packageInfo.targets.compactMap { target -> ProjectDescription.Target? in
-            guard let products = targetToProducts[target.name] else { return nil }
-            return try Target.from(
-                target: target,
-                products: products,
-                packageName: name,
-                packageInfo: packageInfo,
-                packageInfos: packageInfos,
-                packageFolder: path,
-                packageToProject: packageToProject,
-                productTypes: productTypes,
-                baseSettings: baseSettings,
-                targetSettings: targetSettings,
-                platforms: targetToPlatform,
-                minDeploymentTargets: minDeploymentTargets,
-                targetToResolvedDependencies: targetToResolvedDependencies,
-                targetToModuleMap: targetToModuleMap
-            )
-        }
+        let targets = try packageInfo.targets
+            .compactMap { target -> [ProjectDescription.Target]? in
```

Comment:
You can just use `flatMap` instead of `compactMap` + `reduce`

Reply:
I think the `flatMap` for nillable sequence elements is deprecated - or did you mean something else ?
<img width="872" alt="Screenshot 2022-06-26 at 09 53 45" src="https://user-images.githubusercontent.com/12934015/175805036-eb6c8ffe-66bf-43f3-90ac-00c9ca371ebe.png">

Reply:
Sorry I was wrong, modified the code to use `flatMap`